doc/howtos/rdtraining/sprinkles.rst

```diff
@@ -0,0 +1,481 @@
+:ref:`reference/js/widgets`.
+
+.. note::
+
+    **Goal**: at the end of this section, the state of the property is displayed thanks a
+    specific widget:
+
+    .. image:: sprinkles/media/widget.png
+      :align: center
+      :alt: Statusbar widget
+
+    Four states are displayed: New, Offer Received, Offer Accepted and Sold.
+
+When we added fields our models, we (almost) never had to worry about how the field would look like
+in the user interface. For example, a date picker is provided for a ``Date`` field, or a
+``One2many`` field is automatically displayed as a list. Odoo chooses the right 'widget' depending
+on the field type.
+
+However, in some cases, we want a specific representation of a field which can be done thanks to
+the ``widget`` attribute. We already used it for the ``tag_ids`` field when we used the
+``widget="many2many_tags"`` attribute. Indeed, when we don't use it the field is displayed as a
+list.
+
+Each field type has a set of widgets which can be used to fine tune its display. Some widgets also
+take extra options. An exhaustive list can be found in :ref:`reference/js/widgets`.
+
+.. exercise:: Use the status bar widget
+
+    Use the ``statusbar`` widget in order to display the ``state`` of the ``estate.property`` as
+    depicted in the **Goal** of the section.
+
+    Tip: a simple example can be found
+    `here <https://github.com/odoo/odoo/blob/0e12fa135882cd5095dbf15fe2f64231c6a84336/addons/account/views/account_bank_statement_views.xml#L136>`__.
+
+List Order
+==========
+
+**Reference**: the documentation related to this section can be found in
+:ref:`reference/orm/models`.
+
+.. note::
+
+    **Goal**: at the end of this section, all lists are displayed by default in a deterministic
+    order. Property types can be ordered manually.
+
+During the various exercises, we created several list views. However, at no point we specified
+in which order the records had to be listed by default. This is of major importance in many business
+cases. For example, in our real estate module we want to display the highest offers on top of the
+list.
+
+Model
+-----
+
+Odoo provides several ways to provide a default order. The most common way to do it is to define
+the ``_order`` attribute directly on the model. This way, the retrieved records will follow
+a deterministic order which will be consistent in all views, but also when records are searched
+programmatically. Indeed, by default there is no order specified, therefore the records will be
+retrieved in a non-deterministic order depending on PostgreSQL.
+
+The ``_order`` attribute takes a string containing a list of fields which will be used for sorting.
+It will be converted in an order_by_ clause in SQL. For example:
+
+.. code-block:: python
+
+    from odoo import fields, models
+
+    class TestModel(models.Model):
+        _name = "test.model"
+        _description = "Test Model"
+        _order = "id desc"
```

Comment:
```suggestion
        _order = "id desc, description asc"
```

Reply:
But the id is unique, so this won't change anything.

Reply:
Indeed, it was to show a 'list' and an ascending order. My example wasn't good, sorry.